### PR TITLE
Flaky test fix. Query only 1 broker to test quota split

### PR DESCRIPTION
**Pull request by shounakmk219**
## Description

The flaky behaviour was seen for the tests that check for quota across multiple brokers. The queries were made on controller which sometimes distributed queries unevenly across the brokers and ended up exceeding quota on few brokers when it was not supposed to.

### Changes in the PR

1. Query a single broker while testing quota across multiple brokers to avoid uneven query distribution via controller
2. Use the broker debug endpoints to check for quota change propagation.

### CODE REVIEW

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/QueryQuotaClusterIntegrationTest.java [hunk 1]
```diff
@@ -125,12 +132,15 @@ public void testDatabaseQueryQuotaWithTableQueryQuotaWithExtraBroker()
       addQueryQuotaToTableConfig(10);
       // Add one more broker such that quota gets distributed equally among them
       brokerStarter = startOneBroker(2);
+      _brokerBaseApiUrl = LOCAL_HOST + ":" + brokerStarter.getPort();
       // to allow change propagation to QueryQuotaManager
       Thread.sleep(1000);
```
Comment (Jackie-Jiang) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/QueryQuotaClusterIntegrationTest.java:137]: Sleep is not robust, especially in the CI environment where machines have very limited resource. One way to make it more robust is to use `TestUtils.waitForCondition()` and keep trying until it meets the expectation.

**Comment by Jackie-Jiang** [comment]
Any progress on this fix?

**Comment by shounakmk219** [comment]
Hey @Jackie-Jiang  I could not find any direct way of getting the signal that query quota rate limiter change is propagated so right now I am checking for the delay between 2 successful requests and ensure it lies within the expected changed rate (with 20% margin).

**Comment by Jackie-Jiang** [comment]
@klsince Can you help verify the test logic?

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/QueryQuotaClusterIntegrationTest.java [hunk 2]
```diff
@@ -169,14 +193,60 @@ private void runQueries(double qps, boolean shouldFail)
     assertTrue((failCount == 0 && !shouldFail) || (failCount != 0 && shouldFail));
   }
 
+  private void verifyQuotaUpdate(long quotaQps) {
+    String query = "SELECT COUNT(*) FROM " + getTableName();
+    if (quotaQps == 0) {
+      TestUtils.waitForCondition(aVoid -> !executeQueryOnBroker(query).getExceptions().elements().hasNext()
```
Comment (klsince) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/QueryQuotaClusterIntegrationTest.java:199]: looks like the two stmts are same. is this intended to run the query twice back to back?

Comment (klsince) [reply pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/QueryQuotaClusterIntegrationTest.java:199]: perhaps we can add a broker restful API to get runtime quotas for a table, which can help simplify the test here and can be helpful for debugging in prod as well.

Comment (shounakmk219) [reply pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/QueryQuotaClusterIntegrationTest.java:199]: > looks like the two stmts are same. is this intended to run the query twice back to back?

Yes, basically making sure consecutive queries pass which indicated quota is not applied.

> perhaps we can add a broker restful API to get runtime quotas for a table, which can help simplify the test here and can be helpful for debugging in prod as well.

That's a good idea, will use this approach.

Comment (shounakmk219) [reply pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/QueryQuotaClusterIntegrationTest.java:199]: PR to introduce the debug endpoints on broker https://github.com/apache/pinot/pull/13864

**Review by klsince — APPROVED** [review]
LGTM

could you add some PR description about why the test was flaky and how you fixed it briefly

**Pull request merged**